jina/peapods/zmq/__init__.py

```diff
@@ -276,10 +276,13 @@ async def send_message(self, msg: 'Message', sleep: float = 0, **kwargs):
     async def recv_message(self, callback: Callable[['Message'], Union['Message', 'Request']] = None) -> 'Message':
         try:
             msg = await recv_message_async(self.in_sock, **self.send_recv_kwargs)
-            self.bytes_recv += msg.size
             self.msg_recv += 1
-            if callback:
-                return callback(msg)
+            if msg is not None:
+                self.bytes_recv += msg.size
+                if callback:
+                    return callback(msg)
+            else:
+                self.logger.error('Received message is empty.')
```

Comment:
I am not sure about this one. In the end it is "just" better logging information in CI and will most likely not bring any benefit for real flows. Should I leave it in? Or rather catch an `AttributeError`?

Reply:
I think it is fine, however, we need to change the function return to `Optional['Message]`